R/utilities.r

```diff
@@ -668,11 +669,13 @@ with_ordered_restart <- function(expr, .call) {
         ")"
       )
 
-      deprecate_soft0(
-        "3.4.0",
-        I(msg),
-        details = desc
-      )
+      if (inherits(cnd, "vctrs_error_ptype2")) {
```

Comment:
Can you explain about the reasoning for this guard? Why is it suddenly necessary?

Reply:
This was more out of practicality than necessity. Without this, combining two ordered factors would throw 2 warnings, once during `vec_ptype2()` and once during `vec_cast()`. I don't know if there is a situation where `vec_cast()` should warn whereas `vec_ptype2()` shouldn't. See also here: https://github.com/tidyverse/ggplot2/issues/5109#issuecomment-1365831369

Reply:
hmm... maybe keep track of whether the input is two ordered vectors and only in that case have the guard...

Maybe it is overdoing it but since I can't figure out the repercussions I'd rather be on the safe side

Reply:
I too feel uncomfortable about this guard. I think the order of events is as follows:

1. `vec_ptype2(x, y)` searches for a common class to coerce `x` and `y` to. Let's call that `w`.
2. `vec_cast(x, w)` coerces `x` to `w`.
3. `vec_cast(y, w)` coerces `y` to `w`

So if `x` and `y` are ordered factors, `w` will be a factor. The case where we only have two ordered factors is step (1). So, step (2) and (3) will still emit a warning about converting ordered to factors. The ideal situation would be to emit a warning at (1) and then silence (2) and (3). However, the 'track whether x and y are both ordered factors'-rule can't be applied in (2) and (3) because `w` is a regular factor.

I'll puzzle around a bit more to see what might be done, but it'll probably become more complicated than just tracking ordered factors.